Tools/clinic/cpp.py

```diff
@@ -1,7 +1,11 @@
 import re
 import sys
+from typing import NoReturn, Callable
 
-def negate(condition):
+TokenAndCondition = tuple[str, str]
+TokenStack = list[TokenAndCondition]
+
+def negate(condition: str) -> str:
```

Comment:
I guess what one would do here, is to make a "condition" type, and make helper functions like `negate` return `Condition` iso. `str`, right?

Reply:
`typing.NewType` might be a good fit?

Reply:
Maybe; I'll have a look!

Reply:
Hm, I think that might be too picky. I need to cast the return values now.